internal/controller/providers/pxc/applier.go

```diff
@@ -751,53 +753,101 @@ func (p *applier) genPXCBackupSpec() (*pxcv1.PXCScheduledBackup, error) {
 }
 
 func (p *applier) addBackupStorages(
-	backupList *everestv1alpha1.DatabaseClusterBackupList,
+	backups []everestv1alpha1.DatabaseClusterBackup,
+	dataSource *everestv1alpha1.DataSource,
 	storages map[string]*pxcv1.BackupStorageSpec,
 ) error {
-	database := p.DB
-	for _, backup := range backupList.Items {
+	for _, backup := range backups {
 		if _, ok := storages[backup.Spec.BackupStorageName]; ok {
 			continue
 		}
 
-		spec, backupStorage, err := p.genPXCStorageSpec(
-			backup.Spec.BackupStorageName,
-			database.GetNamespace(),
-		)
+		spec, err := p.getStoragesSpec(backup.Spec.BackupStorageName)
 		if err != nil {
-			return errors.Join(err, fmt.Errorf("failed to get backup storage for backup %s", backup.Name))
+			return err
 		}
-
 		storages[backup.Spec.BackupStorageName] = spec
+	}
+	// add the storage from datasource. The restore works without listing the related storage in the pxc config,
+	// however if the storage is insecure, we need to specify it explicitly to set the insecureTLS flag
+	if dataSource != nil {
```

Comment:
I think we should make it:
```go
	if dataSource != nil && (dataSource.DBClusterBackupName != "" || dataSource.BackupSource != nil) {
```

... to avoid problems when data import is specified